src/components/App/App.jsx

```diff
@@ -0,0 +1,104 @@
+  const [storeGame, setStoreGame] = useState({});
+
+  const closeModal = () => setActiveModal("");
+  const navigate = useNavigate();
+  const location = useLocation();
+
+
+  const pageVariants = {
+    initial: { opacity: 0, x: 0 },
+    animate: { opacity: 1, x: 0 },
+    exit: { opacity: 0, x: 0 },
+  };
+
+  const handleLibrarySearch = () =>{
+      /*api request for data*/
+      setLibraryGame({
+          /*hard coded for example usage*/
+          title: "half-life 2",
+          images: [ "https://shared.cloudflare.steamstatic.com/store_item_assets/steam/apps/220/0000001864.600x338.jpg?t=1727742736",
+              "https://shared.cloudflare.steamstatic.com/store_item_assets/steam/apps/220/header.jpg?t=1727742736",
+              "https://shared.cloudflare.steamstatic.com/store_item_assets/steam/apps/220/0000001872.600x338.jpg?t=1727742736",],
+          price: "$24.99",
+          description: "Half Life Description",
+      })
+  }
+  const handleStoreSearch = () =>{
+      /*api request for data*/
+      setStoreGame({
+          /*hard coded for example usage*/
+          title: "Fallout 4",
+          images: [ "https://shared.cloudflare.steamstatic.com/store_item_assets/steam/apps/220/0000001864.600x338.jpg?t=1727742736",
+              "https://shared.cloudflare.steamstatic.com/store_item_assets/steam/apps/220/header.jpg?t=1727742736",
+              "https://shared.cloudflare.steamstatic.com/store_item_assets/steam/apps/220/0000001872.600x338.jpg?t=1727742736",],
+          price: "$59.99",
+          description: "Fallout 4 Description",
+      })
+  }
+
+  return (
+    <librarySelectionContext.Provider value={{librarySelection, setLibrarySelection}}>
+      <activeModalContext.Provider value={{activeModal, setActiveModal, closeModal}}>
+        <div className='app'>
+        {activeModal === "about" && (
+            <motion.div
+            variants={pageVariants}
+            initial="initial"
+            animate="animate"
+            exit="exit"
+            transition={{ duration: 0.5 }}>
+              <About closeModal={closeModal}/> 
+            </motion.div>
+        )}
+          <Header navigate={navigate}/>
+          <AnimatePresence mode="wait">
+            <Routes location={location}> 
+              <Route
+                path="/"
+                element={
+                  <motion.div
+                  initial="initial"
+                  animate="animate"
+                  exit="exit"
+                  variants={pageVariants}
+                  transition={{ duration: 0.5 }}
+                >
+                  <Main navigate={navigate} />
+                </motion.div>
+                }>
+              </Route>
+              <Route
+              path="libraries/*"    
+              element={<Libraries/>
+              }
+              >
+                <Route path="steamstore" element={<GameFinder search={() => handleLibrarySearch} gameInfo={libraryGame}/>}></Route>
```

Comment:
`search={() => handleLibrarySearch} `

This code is misleading. You should either pass `handleLibrarySearch` and `handleStoreSearch` or call them inside the arrow function


```jsx
                <Route path="steamstore" element={<GameFinder search={handleLibrarySearch} gameInfo={libraryGame}/>}></Route>
                <Route path="userlibrary" element={<GameFinder search={handleStoreSearch} gameInfo={storeGame}/>}></Route>

```